packages/bot-web-ui/src/components/load-modal/local.tsx

```diff
@@ -1,21 +1,36 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import classNames from 'classnames';
 import { Button, Icon } from '@deriv/components';
 import { observer, useStore } from '@deriv/stores';
 import { Localize, localize } from '@deriv/translations';
+import { SectionMessage } from '@deriv-com/quill-ui';
+import { botNotification } from 'Components/bot-notification/bot-notification';
+import { notification_message } from 'Components/bot-notification/bot-notification-utils';
 import { useDBotStore } from 'Stores/useDBotStore';
 import LocalFooter from './local-footer';
 import WorkspaceControl from './workspace-control';
 
 const LocalComponent = observer(() => {
     const { ui } = useStore();
-    const { dashboard, load_modal } = useDBotStore();
+    const { dashboard, load_modal, blockly_store } = useDBotStore();
     const { active_tab, active_tour } = dashboard;
-    const { handleFileChange, loaded_local_file, setLoadedLocalFile } = load_modal;
+    const { handleFileChange, loaded_local_file, setLoadedLocalFile, imported_strategy_type, is_open_button_loading } =
+        load_modal;
 
     const file_input_ref = React.useRef<HTMLInputElement>(null);
     const [is_file_supported, setIsFileSupported] = React.useState(true);
     const { is_mobile } = ui;
+    const { is_loading } = blockly_store;
+
+    useEffect(() => {
+        if (loaded_local_file && is_file_supported && imported_strategy_type !== 'pending' && !is_loading) {
+            if (imported_strategy_type === 'old') {
+                botNotification(notification_message.strategy_conversion, undefined, {
+                    closeButton: false,
+                });
+            }
+        }
```

Comment:
```suggestion
        if (loaded_local_file && is_file_supported && imported_strategy_type !== 'pending' && !is_loading && imported_strategy_type === 'old') {
                botNotification(notification_message.strategy_conversion, undefined, {
                    closeButton: false,
                });
        }
```
This nested if is unnecessary. Both of this means the same thing. Other question - do u need this to be run as an effect?

Reply:
Nested if is not necessary for now but kept it for later if we need to handle it differently for `imported_strategy_type = 'new'`. But yes it needs to run as an effect as there are some asynchronous event going on behind the scene (mainly from Blockly) and this snack bar should only show once those are completed. 

Let me remove the nested if.

Reply:
wokehz~ this is resolved @shafin-deriv